### DISCUSSION TITLE
How to create a windhawk-mod from a beginner.

### DISCUSSION THREAD

**Discussion by codezi** [ramensoftware/windhawk-mods#134] [Q&A] [answered]
Hello everyone,
First of all, thank you,@m417z, for your creating amazing mods for Windows.
I want to learn how to create a windhawk-mod from a beginner.
Can you list me some guide to create mod with explorer.exe (which language to learn first, then,...)
Thank you, and sorry for my bad English.

**Comment by m417z** [comment, marked as answer]
Hi, I'm glad that you like Windhawk and its mods!
The language is C++, so I'd start with that. Afterwards, try creating a new mod, look at the short template that customizes Paint, and try to understand it. Start from the comments in the code, which also mention the blog post [X64 Function Hooking by Example](http://kylehalladay.com/blog/2020/11/13/Hooking-By-Example.html) which is a good introduction to the topic.
Also, you're welcome to join [the Discord channel](https://discord.gg/WZgXScMud7), discuss mod development and ask for help.

**Reply by codezi** [reply]
Thank you very much. I will try it.
